src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1319,29 +1591,507 @@ bool HMICapabilitiesImpl::load_capabilities_from_file() {
-                                                     vehicle_type_so);
-      set_vehicle_type(vehicle_type_so);
+
+    if (!json_root_getter.GetInterfaceJsonMember(hmi_interface::vehicle_info)
+             .isNull()) {
+      auto vehicle_info_default_node =
+          json_root_getter.GetMainJsonMember(hmi_interface::vehicle_info);
+      auto vehicle_info_cache_node =
+          json_root_getter.GetCachedJsonMember(hmi_interface::vehicle_info);
+      JsonCapabilitiesGetter json_vehicle_info_getter(vehicle_info_default_node,
+                                                      vehicle_info_cache_node);
+      auto vehicle_type_node = json_vehicle_info_getter.GetJsonMember(
+          hmi_response::vehicle_type,
+          hmi_apis::FunctionID::VehicleInfo_GetVehicleType,
+          requests_required_for_capabilities_);
+      if (!vehicle_type_node.isNull()) {
+        smart_objects::SmartObject vehicle_type_so;
+        formatters::CFormatterJsonBase::jsonValueToObj(vehicle_type_node,
+                                                       vehicle_type_so);
+        set_vehicle_type(vehicle_type_so);
+      }
     }  // VehicleType end
   } catch (...) {
     return false;
   }
   return true;
 }
 
+hmi_apis::Common_Language::eType
+HMICapabilitiesImpl::GetActiveLanguageForInterface(
+    const std::string& interface_name) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (hmi_interface::ui == interface_name) {
+    return active_ui_language();
+  }
+  if (hmi_interface::vr == interface_name) {
+    return active_vr_language();
+  }
+  if (hmi_interface::tts == interface_name) {
+    return active_tts_language();
+  }
+  return hmi_apis::Common_Language::INVALID_ENUM;
+}
+
+std::set<hmi_apis::FunctionID::eType>
+HMICapabilitiesImpl::GetRequestsRequiredForCapabilities() const {
+  return requests_required_for_capabilities_;
+}
+
+void HMICapabilitiesImpl::UpdateRequestsRequiredForCapabilities(
+    hmi_apis::FunctionID::eType requested_interface) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (app_mngr_.IsHMICooperating()) {
+    LOG4CXX_DEBUG(logger_,
+                  "Remove from default initialized capabilities skipped, "
+                  "because hmi_cooperating equal true already");
+    return;
+  }
+
+  RemoveFromRequestsRequiredForCapabilities(requested_interface);
+  if (requests_required_for_capabilities_.empty()) {
+    app_mngr_.SetHMICooperating(true);
+  }
+}
+
+bool HMICapabilitiesImpl::MatchesCCPUVersion(
+    const std::string& ccpu_version) const {
+  return ccpu_version_ == ccpu_version;
+}
+
+void HMICapabilitiesImpl::OnSoftwareVersionReceived(
+    const std::string& ccpu_version) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (MatchesCCPUVersion(ccpu_version)) {
+    LOG4CXX_DEBUG(logger_, "Software version not changed");
+    if (requests_required_for_capabilities_.empty()) {
+      app_mngr_.SetHMICooperating(true);
+    }
+    app_mngr_.RequestForInterfacesAvailability();
+    return;
+  }
```

Comment:
what about if they do not match? The proposal specifies to remove the cache file and more...